cpp/tests/ast/transform_tests.cpp

```diff
@@ -343,6 +343,74 @@ TEST_F(TransformTest, ImbalancedTreeArithmeticDeep)
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, result->view(), verbosity);
 }
 
+TEST_F(TransformTest, DeeplyNestedArithmeticLogicalExpression)
+{
+  // Test logic for deeply nested arithmetic and logical expressions.
+  constexpr int64_t kLeftDepthLevel  = 100;
+  constexpr int64_t kRightDepthLevel = 75;
+
+  auto generate_ast_expr = [](int64_t depth_level,
+                              cudf::ast::column_reference col_ref,
+                              cudf::ast::ast_operator root_operator,
+                              bool nested_left_tree) {
+    auto expressions                                          = std::list<cudf::ast::operation>();
+    std::vector<cudf::ast::ast_operator> arithmetic_operators = {cudf::ast::ast_operator::ADD,
+                                                                 cudf::ast::ast_operator::SUB};
+
+    auto op = arithmetic_operators[rand() % arithmetic_operators.size()];
```

Comment:
For consistent results could we seed the RNG? We can always run multiple times with different seeds if we like, and I do see the appeal of essentially fuzz testing, OTOH intermittent failures make CI quite painful.

Reply:
I used different operator for each side, + for left subtree and - for right subtree, so that the code comments are clear too.

Reply:
Sounds good.